Project/DAQ_System/Program/Src/Application/FileSystem/fat_fs.cpp

```diff
@@ -0,0 +1,120 @@
+#include "fat_fs.hpp"
+
+namespace application {
+
+FatFs::FatFs(char logical_drive_path[kMaxDrivePathLength], FATFS& file_system, FIL& file)
+  : logical_drive_path_(logical_drive_path), file_system_(file_system), file_(file) { }
+
+FatFs::~FatFs() { }
+
+bool FatFs::Mount () {
+	fs_result_ = f_mount(&file_system_, logical_drive_path_, kForceMount);
+	return EvaluateResult(ErrorFlags::Mounting);
+}
+
+bool FatFs::Unmount () {
+	fs_result_ = f_mount(NULL, logical_drive_path_, kForceMount);
+	return EvaluateResult(ErrorFlags::Unmounting);
+}
+
+bool FatFs::FileExists(char* path) {
+	fs_result_ = f_stat(path, &file_info_);
+	return EvaluateResult(ErrorFlags::FileNotFound);
+}
+
+bool FatFs::CreateFile(char* path) {
+	return FileDoesNotExist(path) &&
+			OpenFile(path, kEmtpyFileAndWriteMode) &&
+			CloseFile();
+}
+
+bool FatFs::OpenFile(char* path, const char* access_mode) {
+	if (access_mode == nullptr) {
+		return false;
+	}
+
+	BYTE fatfs_mode = '\0';
+
+	// Convert POSIX compliant flag to FATFS flag
+	if (access_mode == kReadOnlyMode) {
+		fatfs_mode = FA_READ;
+	} else if (strcmp(access_mode, kReadWriteMode) == 0) {
+		fatfs_mode =  FA_READ | FA_WRITE;
+	} else if (strcmp(access_mode, kEmtpyFileAndWriteMode) == 0) {
+		fatfs_mode = FA_CREATE_ALWAYS | FA_WRITE;;
+	} else if (strcmp(access_mode, kEmptyFileAndReadWriteMode) == 0) {
+		fatfs_mode = FA_CREATE_ALWAYS | FA_WRITE | FA_READ;
+	} else if (strcmp(access_mode, kAppendFileAndWriteMode) == 0) {
+		fatfs_mode = FA_OPEN_APPEND | FA_WRITE;
+	} else if (strcmp(access_mode, kAppendFileAndReadWriteMode) == 0) {
+		fatfs_mode = FA_OPEN_APPEND | FA_WRITE | FA_READ;
+	} else if (strcmp(access_mode, kNewFileAndWriteMode) == 0) {
+		fatfs_mode = FA_CREATE_NEW | FA_WRITE;
+	} else if (strcmp(access_mode, kNewFileAndReadWriteMode) == 0) {
+		fatfs_mode = FA_CREATE_NEW | FA_WRITE | FA_READ;
+	} else {
+		return false;
+	}
```

Comment:
Such a shame that switch statements are not allowed with `char` data... 😥

Reply:
Switch statements have really grown on me